### ISSUE TITLE
Build fails with Clang 13

### ISSUE THREAD

**Issue by awvwgk** [qcscine/utilities#4]
Trying to build version 5.0.0 fails with clang 13 on OSX with the following error

```
[156/287] Building CXX object src/Utils/CMakeFiles/UtilsOS.dir/Utils/Scf/OrbitalPerturbation/RandomOrbitalMixer.cpp.o
FAILED: src/Utils/CMakeFiles/UtilsOS.dir/Utils/Scf/OrbitalPerturbation/RandomOrbitalMixer.cpp.o 
$BUILD_PREFIX/bin/x86_64-apple-darwin13.4.0-clang++ -DBOOST_ALL_NO_LIB -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_SYSTEM_DYN_LINK -DEIGEN_MATRIX_PLUGIN=\"libirc/plugins/eigen/Matrix_initializer_list.h\" -DEIGEN_USE_BLAS -DHAVE_EIGEN3 -DUtilsOS_EXPORTS -I$SRC_DIR/src/Utils -I$SRC_DIR/_build/irc-src/include -isystem $PREFIX/include/Scine -isystem $PREFIX/include/eigen3 -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -stdlib=libc++ -fvisibility-inlines-hidden -fmessage-length=0 -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/scine-utilsos-5.0.0 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix -Wall -Wextra -Wpedantic -O3 -DNDEBUG -isysroot /Applications/Xcode_12.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk -mmacosx-version-min=10.9 -fPIC -fopenmp=libomp -std=gnu++17 -MD -MT src/Utils/CMakeFiles/UtilsOS.dir/Utils/Scf/OrbitalPerturbation/RandomOrbitalMixer.cpp.o -MF src/Utils/CMakeFiles/UtilsOS.dir/Utils/Scf/OrbitalPerturbation/RandomOrbitalMixer.cpp.o.d -o src/Utils/CMakeFiles/UtilsOS.dir/Utils/Scf/OrbitalPerturbation/RandomOrbitalMixer.cpp.o -c $SRC_DIR/src/Utils/Utils/Scf/OrbitalPerturbation/RandomOrbitalMixer.cpp
In file included from $SRC_DIR/src/Utils/Utils/Scf/OrbitalPerturbation/RandomOrbitalMixer.cpp:9:
$SRC_DIR/src/Utils/Utils/Scf/OrbitalPerturbation/UniqueRandomNumbersGenerator.h:65:8: error: no member named 'random_shuffle' in namespace 'std'
  std::random_shuffle(shuffler.begin(), shuffler.end());
  ~~~~~^
1 error generated.
```

System details

```
     active environment : base
    active env location : /Users/runner/miniforge3
            shell level : 1
       user config file : /Users/runner/.condarc
 populated config files : /Users/runner/miniforge3/.condarc
          conda version : 4.12.0
    conda-build version : 3.21.9
         python version : 3.9.13.final.0
       virtual packages : __osx=10.15.7=0
                          __unix=0=0
                          __archspec=1=x86_64
       base environment : /Users/runner/miniforge3  (writable)
      conda av data dir : /Users/runner/miniforge3/etc/conda
  conda av metadata url : None
           channel URLs : https://conda.anaconda.org/conda-forge/osx-64
                          https://conda.anaconda.org/conda-forge/noarch
          package cache : /Users/runner/miniforge3/pkgs
                          /Users/runner/.conda/pkgs
       envs directories : /Users/runner/miniforge3/envs
                          /Users/runner/.conda/envs
               platform : osx-64
             user-agent : conda/4.12.0 requests/2.27.1 CPython/3.9.13 Darwin/19.6.0 OSX/10.15.7
                UID:GID : 501:20
             netrc file : None
           offline mode : False
```

See [build log](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=529465&view=logs&j=58ac6aab-c4bc-5de2-2894-98e408cc8ec9&t=933f325c-924e-533d-4d95-e93b5843ce8b)

---

Looking at [`std::random_shuffle`](https://en.cppreference.com/w/cpp/algorithm/random_shuffle) it seems like the current usage is removed in C++17.

**Comment by awvwgk** [comment]
Here is the patch I'm using to fix the build. Feel free to include it for the next release.

```patch
diff --git a/src/Utils/Utils/Scf/OrbitalPerturbation/UniqueRandomNumbersGenerator.h b/src/Utils/Utils/Scf/OrbitalPerturbation/UniqueRandomNumbersGenerator.h
index 04cddc3..3f10aa5 100644
--- a/src/Utils/Utils/Scf/OrbitalPerturbation/UniqueRandomNumbersGenerator.h
+++ b/src/Utils/Utils/Scf/OrbitalPerturbation/UniqueRandomNumbersGenerator.h
@@ -12,6 +12,7 @@
 #include <cassert>
 #include <numeric>
 #include <vector>
+#include <random>
 
 namespace Scine {
 namespace Utils {
@@ -62,7 +63,11 @@ std::vector<IntegerType> UniqueRandomNumbersGenerator<IntegerType>::generate(uns
 
   std::vector<IntegerType> shuffler(max_ - min_ + 1);
   std::iota(shuffler.begin(), shuffler.end(), min_);
-  std::random_shuffle(shuffler.begin(), shuffler.end());
+
+  std::random_device rd;
+  std::mt19937 g(rd());
+  std::shuffle(shuffler.begin(), shuffler.end(), g);
+
   return {shuffler.begin(), shuffler.begin() + N};
 }
 
```

**Comment by weymutht** [comment]
Thank you very much for providing this patch. I'll try it out on our machines. If it works for us, we'll gladly incorporate it.

**Comment by reiher-research-group** [comment]
This is resolved with Utilities 6.0.0, released today.
